pkg/storage/ingest/fetcher.go

```diff
@@ -312,6 +324,10 @@ func (r *concurrentFetchers) PollFetches(ctx context.Context) (kgo.Fetches, cont
 	case <-ctx.Done():
 		return kgo.Fetches{}, ctx
 	case f := <-r.orderedFetches:
+		// At this point, we're guaranteed that the records are not going to be with us anymore.
+		tr := r.bufferedFetchedRecords.Sub(int64(len(f.FetchPartition.Records)))
+		level.Debug(r.logger).Log("msg", "remove buffered fetched records", "num_records", len(f.FetchPartition.Records), "total_records", tr)
```

Comment:
[nit] I typically recommend to not keep debug logs for hot path. They have a computational cost even when debug level is disabled.

_Same applies below._

Reply:
Oh, TIL.

Reply:
Done in https://github.com/grafana/mimir/pull/9855/commits/43f6386c4e650ca5b22d651d4ba0a7cb60d9672d